src/VisualStudio/Core/Def/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -570,10 +577,10 @@ public ITableEntriesSnapshot GetCurrentSnapshot()
                     // Otherwise return the appropriate list based on how we're currently
                     // grouping.
                     var entries = _cleared
-                        ? ImmutableList<Entry>.Empty
+                        ? []
                         : _currentlyGroupingByDefinition
-                            ? EntriesWhenGroupingByDefinition
-                            : EntriesWhenNotGroupingByDefinition;
+                            ? EntriesWhenGroupingByDefinition.ToImmutableArray()
+                            : EntriesWhenNotGroupingByDefinition.ToImmutableArray();
```

Comment:
here we make a snapshot to actually put into TableEntriesSnapshot.  But this should be ok the vast majority of hte time.  We're only updating this snapshot at most 4 times a second.  So the amount of copies made by this is very small (didn't show up in any measurements).  And the wins we get by just working with List vs an ImmutableList (which internally is an AVL tree) is substantive.